### ISSUE TITLE
Some error in runtime

### ISSUE THREAD

**Issue by ytkmy5555** [YyzHarry/imbalanced-regression#18]

**Comment by ytkmy5555** [comment]
Hi @YyzHarry When I try to run your code, the system prompts me with the following message: overwrite previous folder: checkpoint\agedb_resnet50_adam_l1_0.001_256 ? [Y/n] :
If my answer is Y, the error message is:
![image](https://user-images.githubusercontent.com/90555756/152504499-868111dc-ee67-4b92-b22d-d1b0ae61fbb1.png)

**Comment by ytkmy5555** [comment]
Hi @YyzHarry,
I want to use your code of LDS to solve my problem, but I have some problems in reproducing your code：
import os
import logging
import numpy as np
from scipy.ndimage import convolve1d
from torch.utils import data
import pandas as pd

from utils import get_lds_kernel_window

import seaborn as sns
import matplotlib.pyplot as plt


def _prepare_weights(labels, reweight, max_target=121, lds=False, lds_kernel='gaussian', lds_ks=5, lds_sigma=2):
    assert reweight in {'none', 'inverse', 'sqrt_inv'}
    assert reweight != 'none' if lds else True, \
        "Set reweight to \'sqrt_inv\' (default) or \'inverse\' when using LDS"

    value_dict = {x: 0 for x in range(max_target)}
    # labels = self.df['age'].values
    for label in labels:
        value_dict[min(max_target - 1, int(label))] += 1
    if reweight == 'sqrt_inv':
        value_dict = {k: np.sqrt(v) for k, v in value_dict.items()}
    elif reweight == 'inverse':
        value_dict = {k: np.clip(v, 5, 1000) for k, v in value_dict.items()}  # clip weights for inverse re-weight
    num_per_label = [value_dict[min(max_target - 1, int(label))] for label in labels]
    if not len(num_per_label) or reweight == 'none':
        return None
    print(f"Using re-weighting: [{reweight.upper()}]")

    if lds:
        lds_kernel_window = get_lds_kernel_window(lds_kernel, lds_ks, lds_sigma)
        print(f'Using LDS: [{lds_kernel.upper()}] ({lds_ks}/{lds_sigma})')
        smoothed_value = convolve1d(
            np.asarray([v for _, v in value_dict.items()]), weights=lds_kernel_window, mode='constant')
        num_per_label = [smoothed_value[min(max_target - 1, int(label))] for label in labels]

    weights = [np.float32(1 / x) for x in num_per_label]
    scaling = len(weights) / np.sum(weights)
    weights = [scaling * x for x in weights]
    return weights


def main():
    data = pd.read_csv(r"./data/agedb.csv", header=None, delimiter=',', low_memory=False)
    labels = data.values[1:][:,:1]
    weights1 = _prepare_weights(labels, reweight='sqrt_inv', lds=True, lds_kernel='gaussian', lds_ks=5, lds_sigma=2)
    sns.distplot(labels, hist=True, axlabel='anom', color='r',
                 kde_kws={"label": "pre"})
    sns.distplot(weights1, hist=True, axlabel='anom',color='b',
                 kde_kws = {"label":"fur"})
    plt.show()
    return

if __name__ == '__main__':
    print('Start!!!!')

    main()
    print('End!!!!')
    os._exit(0)

The  result are：
(1)(2)
![image](https://user-images.githubusercontent.com/90555756/152627095-be5db078-4295-4d8e-9d6a-a5af9c1037c6.png)

![image](https://user-images.githubusercontent.com/90555756/152627076-b69e7268-302f-4ba1-b99f-0eb8c433f6a9.png)
Which step was wrong？ I would appreciate it if you could give me some guidance.

**Comment by kaiwenzha** [comment]
Hi @ytkmy5555 I am confused about your question. It seems that you visualize the label distribution and its LDS weight distribution respectively. Then during training, you just need to apply the weights when calculating the loss.
https://github.com/YyzHarry/imbalanced-regression/blob/055a7b3804bbaf903ed25a55c11ab8acc6e142e1/agedb-dir/train.py#L246-L255
https://github.com/YyzHarry/imbalanced-regression/blob/055a7b3804bbaf903ed25a55c11ab8acc6e142e1/agedb-dir/loss.py#L5-L10
